### PR TITLE
Regex to catch bad data mount

### PR DESCRIPTION
**Pull request by ReDeiPirati**
- [[cli] crash in processing data mount ids](https://app.asana.com/0/817307208669065/750539112198233)

```bash
(cli-venv) pirate:cli-test pirate$ floyd run --data https://www.floydhub.com/pirate/datasets/my-data:
Argument 'https://www.floydhub.com/pirate/datasets/my-data:' doesn't match any recognized pattern:
        floyd run --data <dataset_name>
	floyd run --data <dataset_name>:<mounting_point>
	floyd run --data <namespace>/[projects|dataset]/<dataset_or_project_name>
	floyd run --data <namespace>/[projects|dataset]/<dataset_or_project_name>
	floyd run --data <namespace>/[projects|dataset]/<dataset_or_project_name>/<version>
	floyd run --data <namespace>/[projects|dataset]/<dataset_or_project_name>/<version>:<mounting_point>
	floyd run --data <data_id> (DEPRECATED)
	floyd run --data <data_id>:<mounting_point> (DEPRECATED)

 Note: Argument can contain only alphanumeric and - chars
```

### CODE REVIEW

**Comment by ReDeiPirati** [comment]
It seems that the tests have to be updated.

File: floyd/cli/run.py [hunk 1]
```diff
@@ -51,6 +64,18 @@ def process_data_ids(data_ids):
...
+            sys.exit(("Argument '%s' doesn't match any recognized pattern:\n"
+                      "\tfloyd run --data <dataset_name>\n"
+                      "\tfloyd run --data <dataset_name>:<mounting_point>\n"
+                      "\tfloyd run --data <namespace>/[projects|datasets]/<dataset_or_project_name>\n"
+                      "\tfloyd run --data <namespace>/[projects|datasets]/<dataset_or_project_name>:<mounting_point>\n"
+                      "\tfloyd run --data <namespace>/[projects|datasets]/<dataset_or_project_name>/<version>\n"
+                      "\tfloyd run --data <namespace>/[projects|datasets]/<dataset_or_project_name>/<version>:<mounting_point>\n"
+                      "\tfloyd run --data <data_id> (DEPRECATED)\n"
+                      "\tfloyd run --data <data_id>:<mounting_point> (DEPRECATED)\n"
+                      "\n Note: Argument can contain only alphanumeric and - chars"
```
[the remark this answers is not in the document]
Comment (ReDeiPirati) [reply floyd/cli/run.py]: The dash `-` is allowed too, I will add this between parenthesis

File: floyd/cli/run.py [hunk 2]
```diff
@@ -51,6 +64,18 @@ def process_data_ids(data_ids):
...
+                      ) % data_name_or_id)
```
[the remark this answers is not in the document]
Comment (ReDeiPirati) [reply floyd/cli/run.py]: 👍

File: floyd/cli/run.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (narenst) [reply floyd/cli/run.py]: Ah, sorry I missed that. Yeah that works.

**Pull request merged**